internal/command/commands/component.go

```diff
@@ -0,0 +1,91 @@
+package commands
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+
+	"github.com/dream11/odin/pkg/utils"
+	"github.com/dream11/odin/internal/backend"
+	"github.com/dream11/odin/api/component"
+)
+
+var componentClient backend.Component
+
+type Component command
+
+func (c *Component) Run(args []string) int {
+	// Define flag set
+	flagSet := flag.NewFlagSet("flagSet", flag.ContinueOnError)
+	name:= flagSet.String("name", "", "name of the component")
+	serviceName:= flagSet.String("service", "", "name of the service in which the component is deployed")
+	envName:= flagSet.String("env", "", "name of the environment in which the service is deployed")
+	operation:= flagSet.String("operation", "", "name of the operation to performed on the component")
+	options:= flagSet.String("options", "", "options of the operation in JSON format")
+
+	err := flagSet.Parse(args)
+	if err != nil {
+		c.Logger.Error("Unable to parse flags! " + err.Error())
+		return 1
+	}
+
+	if c.Operate {
+		if *envName == "" {
+			*envName = utils.FetchKey(ENV_NAME_KEY)
+		}
+		emptyParameters := emptyParameters(map[string]string{"--name": *name, "--service": *serviceName, "--env": *envName, "--operation": *operation, "--options": *options})
+		if len(emptyParameters) == 0 {
+			var optionsJson interface{}
+			err = json.Unmarshal([]byte(*options), &optionsJson)
+			if err != nil {
+				c.Logger.Error("Unable to parse options JSON " + err.Error())
+				return 1
+			}
+
+			data:= component.OperateComponentRequest{
+				Data: component.Data{
+					EnvName: *envName,
+					ServiceName: *serviceName,
+					Operations: []component.Operation{
```

Comment:
why operations list? can the user provide multiple operations for the same component in the same command? what if the operations are dependent on eachother?

Reply:
currently only one operation is allowed at a time but in future we may allow multiple operations then we no need to change the contract again